components/list/demo/list-item-sample.js

```diff
@@ -10,55 +11,25 @@ class ListItemSample extends ListItemCheckboxMixin(LitElement) {
 		};
 	}
 
-	// TODO: Placeholder grid styles to be replaced with list-item-generic
 	static get styles() {
-		return [ (super.styles ? super.styles : css``), css`
-			.list-grid {
-				display: grid;
-				grid-template-columns:
-					[start outside-control-start] 40px
-					[control-start outside-control-end] 40px
-					[control-end content-start] auto
-					[content-end actions-start] auto
-					[end actions-end];
-			}
-			[slot="content"]{
-				grid-column: content-start / content-end;
-				grid-row: 1 / 2;
-			}
-			[slot="control"] {
-				grid-column: control-start / control-end;
-				grid-row: 1 / 2;
-			}
-			[slot="control-action"] {
-				grid-column: control-start / end;
-				grid-row: 1 / 2;
-				z-index: 2;
-				cursor: pointer;
-			}
-			:host([disabled]) [slot="control-action"] {
-				cursor: default;
-			}
-			::slotted([slot="actions"]) {
-				grid-column: actions-start / actions-end;
-				grid-row: 1 / 2;
-				z-index: 4;
-			}
-		`];
+		return [ (super.styles ? super.styles : css``)];
 	}
+
 	render() {
 		return html`
-			<div class="list-grid">
+			<d2l-list-item-generic>
```

Comment:
```
classMapRender = ['d2l-list-item-area-control': true] //default value but overridable
classMapAction = ['d2l-list-item-area-control-action': true] //default value but overridable
<d2l-list-item-generic>
      ${renderCheckbox(classMapRender, classMapAction)} 
</d2l-list-item-generic>
```
Then magic happens and it just works.

Reply:
Granted the way you are doing it has similar results.

The reason I wanted the generic list host to define the grid is so the user could override it if they wanted too. Which probably makes things difficult with focus order stuff. :D Sorry new information that might not be required.

Reply:
My thought was that if the user wants to override the default layout, they would just need to implement a new version of `list-item-generic` to do so.

I guess the question is whether we want to force focus order with slots or not. If not, I'll need to think of another solution for that problem.

Reply:
I would prefer that no one makes a new version of `list-item-generic`

Reply:
That makes sense. I think I'm not super clear on the magic that's supposed to happen in the `renderCheckbox` method. Before, I had it so that the checkbox control and action areas were rendered automagically in that method, but @dbatiste mentioned that it would be beneficial for users implementing a new list item component to have separate methods for doing that. 

Are you saying that the user would have control of the two areas via the class maps fed to the method? And if they wanted to change `d2l-list-item-area-control` to `some-custom-class` they would do that through the class map?

How is this different from a user just changing their custom component itself, e.g.,
```
<div class="some-custom-control" slot="control">${ this._renderCheckbox() }</div>
<div class="some-custom-action" slot="control-action">${ this._renderCheckboxAction() }</div>
```
or targetting with css?

Reply:
I don't think it is different.

Reply:
As long as `nothing` works.

Reply:
The current way is probably more intuitive which is better.

Reply:
I will probably always be a fan of more intuitive vs. less markup <- (someone quote this for me in 2025)

But I am totally cool with whatever is easiest for our users, in this case developers.

Reply:
More intuitive is better for devs so I say leave it the way you did it.